app/src/main/java/com/android/streetworkapp/ui/navigation/InfoDialogs.kt

```diff
@@ -0,0 +1,137 @@
+package com.android.streetworkapp.ui.navigation
+
+import android.util.Log
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import com.android.streetworkapp.ui.utils.CustomDialog
+import com.android.streetworkapp.ui.utils.DialogType
+
+/**
+ * This class manages the InfoDialogs that are displayed when the user clicks on the info action. It
+ * is responsible for displaying the correct dialog for the current screen.
+ *
+ * @param showDialog - The display state of the InfoDialogs.
+ * @param screenName - The name of the current screen.
+ * @param topAppBarManager - The TopAppBarManager instance (used to set the action callback).
+ */
+class InfoDialogManager(
+    private val showDialog: MutableState<Boolean>,
+    private val screenName: MutableState<String?>,
+    private val topAppBarManager: TopAppBarManager?
+) {
+  private val screens = getScreens()
+
+  // Mapping Screen to InfoDialog
+  private val infoDialogs: MutableMap<String, InfoDialog> =
```

Comment:
I think it would be a good idea to add a `TODO` here to explain that this must be implemented in the future via `strings.xml`. I would even open an issue explaining the problem in order to keep track of the problem. What do you think of it?

Reply:
I agree with you, it would ensure that we do not forget about the refactoring !

Reply:
Thank you for adding the TODO! Please just open an Issue with the **Refactor** tag so we can track this problem in the future.

Reply:
You can access it here : https://github.com/SwEnt-Group8/Street-work-app/issues/169